chainer/dataset/convert.py

```diff
@@ -0,0 +1,114 @@
+import numpy
+import six
+
+from chainer import cuda
+
+
+def concat_examples(batch, device=None, padding=None):
+    """Concatenates a list of examples into array(s).
+
+    Dataset iterator yields a list of examples. If each example is an array,
+    this function concatenates them along the newly-inserted first axis (called
+    `batch dimension`) into one array. The basic behavior is same for examples
+    consisting of multiple arrays, i.e., corresponding arrays of all examples
+    are concatenated.
+
+    For instance, consider each example consists of two arrays ``(x, y)``.
```

Comment:
To be precise, this is not an example but a specification of this function. In my personal opinion, specifications should be determined without examples. But it is OK not to change the description because it is somewhat the matter of tastes.